packages/foam-vscode/src/services/variable-resolver.ts

```diff
@@ -55,6 +57,7 @@ export function findFoamVariables(templateText: string): string[] {
 
 export class Resolver {
   promises = new Map<string, Thenable<string>>();
+  foamTitle: string | null = null;
```

Comment:
Why do we need to treat the title as a special case?
Could we, when resolving the slug, simply go through the resolution of the title?

Reply:
I tried using `resolveFoamTitle()` for the slug as well, but the input box would show twice or cancel the second one and not replace the variable `FOAM_TITLE_SLUG` in the template.

I can look into refactoring so the `foamTitle` property isn't required though.

Reply:
have you tried `this.resolve('FOAM_TITLE')`?

Reply:
I'll give `this.resolve('FOAM_TITLE')` a try this evening. Thanks for your patience!